### PR TITLE
[web] Detect when the mouseup occurs outside of window

### PR DESCRIPTION
**Pull request by mdebbar**
Fixes https://github.com/flutter/flutter/issues/52827

### CODE REVIEW

File: lib/web_ui/lib/src/engine/pointer_binding.dart [hunk 1]
```diff
@@ -412,11 +423,15 @@ class _PointerAdapter extends _BaseAdapter with _WheelEventListenerMixin {
     }
   }
 
-  void _addPointerEventListener(String eventName, _PointerEventListener handler) {
+  void _addPointerEventListener(
+    String eventName,
+    _PointerEventListener handler, {
+    bool acceptOutsideGlasspane = false,
```
Comment (ferhatb) [inline lib/web_ui/lib/src/engine/pointer_binding.dart:429]: Please add comment on parameter

File: lib/web_ui/lib/src/engine/pointer_binding.dart [hunk 2]
```diff
@@ -196,8 +204,11 @@ abstract class _BaseAdapter {
       }
     };
     _listeners[eventName] = loggedHandler;
-    glassPaneElement
-        .addEventListener(eventName, loggedHandler, true);
+    // We have to attach the event listener on the window instead of the
+    // glasspane element. That's because "up" events that occur outside the
+    // browser are only reported on window, not on DOM elements.
+    // See: https://github.com/flutter/flutter/issues/52827
+    html.window.addEventListener(eventName, loggedHandler, true);
```
Comment (ferhatb) [inline lib/web_ui/lib/src/engine/pointer_binding.dart:211]: have you verified this on Firefox/safari as well ?

Comment (mdebbar) [reply lib/web_ui/lib/src/engine/pointer_binding.dart:211]: Good point. I'll verify.

Comment (yjbanov) [reply lib/web_ui/lib/src/engine/pointer_binding.dart:211]: There has been back'n'forth about global event attachment. Can you please check the history to make sure this is not regressing some corner cases, such as a11y, platform views, iframes?

Comment (mdebbar) [reply lib/web_ui/lib/src/engine/pointer_binding.dart:211]: Tested in Firefox and Safari. Works like a charm!

**Pull request merged**